src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp

```diff
@@ -239,43 +256,47 @@ void PrettyBlockOutputFormat::writeChunk(const Chunk & chunk, PortKind port_kind
     }
 
     /// Names
-    writeCString(grid_symbols.bold_bar, out);
-    writeCString(" ", out);
-    for (size_t i = 0; i < num_columns; ++i)
+    auto write_names = [&]() -> void
```

Comment:
This part of the code can be reused in the footer so I have made it a lambda which gets called again later in the code if the setting is true.